### ISSUE TITLE
Change Network discovery

### ISSUE THREAD

**Issue by realpaaao** [vultisig/vultisig-android#313]
We had a discussion in Discord that we should set default network discovery to relay service and if not possible to discover, give the option to switch to local network.
This will improve UX flow due to less options and confusion.

**Comment by aminsato** [comment]
In #358  I tried to connect via relay, if it fails (if no connectivity) it will automatically switch to local network.
However I

**Comment by johnnyluo** [comment]
I think we should keep the network prompt , but default to relay server ,  still give user the ability to switch back to WIFI / HOTSPOT mode

**Comment by realpaaao** [comment]
I would suggest we add an extra screen where it prompts that no device was detected over relay server and that the user should switch to local network. If user selects switch, prompt pairing screen with QR for local network.
something like that?
![image](https://github.com/vultisig/vultisig-android/assets/167348323/3b9910af-97c3-4336-bc0b-db0d2d8de5d1)

**Comment by realpaaao** [comment]
Maybe we could add a logic that when the initiating device is on pairing screen a timer starts running and after a set time, maybe 1-2 minutes a prompt pops up "No device detected want to switch to local network ?" and change the `continue` button to `Switch to local network`.

something like that:
![image](https://github.com/vultisig/vultisig-android/assets/167348323/81e097fe-c53a-45f5-a6e3-a19d6a239b40)

**Comment by johnnyluo** [comment]
@realpaaao  I have to disagree with this ,  the reason we implement a local wifi setting(local network option) is because we would like user to try to use their local network without talking to cloud when setting up vault and keysign.   That's why we implement it at first place.  Keep everything local is more safer to talking to cloud. 

But now ,we have relay server working ,  given the message is encrypted , and it is secure , so it looks like it is more reliable,  but we should still  keep local network option to user , and encourage to use it , only use relay server as their fallback. 

This change hide the local option at first , only when show up when relay is not working ,  for users they don't trust relay server , and don't want to use it , this make their life much harder. 

So please don't remove the network prompt from this screen

**Comment by realpaaao** [comment]
@johnnyluo I agree with you on the safety aspect. In terms of UX and automation it would be better with less buttons. 
The timeout was just an idea I had, because like you said, it's hard to know why it can't detect the pairing devices. 

Could you be so kind and respond @jpthor on this matter in the [Discord thread](https://discord.com/channels/1203844257220395078/1247991735603429497/1248039305130283108) and share your thoughts?

**Comment by johnnyluo** [comment]
If we want have less buttons , then we could make the section can be expand / collapse

**Comment by realpaaao** [comment]
@johnnyluo added the old layout again. will ping in Discord for expand / collapse design.
@aminsato I updated figma again,m could you add the network options again as soon as we settled for design in discord?
